File: Week 6/cash.py
class RangeError(Exception):
    # define error if value out of range
    pass


def main():
    # populate change
    change = get_change()

    # populate coins
    coins = get_coins(change)

    # print
    print(coins)


def get_coins(change):
    # initialize coins
    coins = 0

    # calculate quarters
    quarters = int(change / 0.25)
    change = change % 0.25
    coins += quarters

    # calculate dimes
    dimes = int(change / 0.10)
    change = change % 0.10
    coins += dimes

    # calculate nickels
    nickels = int(change / 0.05)
    change = change % 0.05
    coins += nickels

    # calculate pennies
    pennies = int(change / 0.01)
    change = change % 0.01
    coins += pennies

    return coins


def get_change():
    while True:
        try:
            # get user input
            change = float(input("Change owed: "))

            # evaluate input against acceptable range
            if change < 0:
                raise RangeError

        # check value
        except ValueError:
            print("Not a float!")

        # check if in range
        except RangeError:
            print("Not within range!")

        else:
            return change


main()

File: Week1/Cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("Number of coins: %i\n", coins);
}

//initialize input

int get_cents(void)
{
    int input;
    do
    {
        //prompt user for input
        input = get_int("Got any cents?: ");
    }
    //while input < 0
    while (input < 0);

    return input;
}

int calculate_quarters(int cents)
{
    int quarters;

    quarters = cents / 25;

    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes;

    dimes = cents / 10;

    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels;

    nickels = cents / 5;

    return nickels;
}

int calculate_pennies(int cents)
{
    int pennies;

    pennies = cents / 1;

    return pennies;
}
